content/blog/cloud-native-secret-management-with-pulumi-esc-and-external-secrets-operator/index.md

```diff
@@ -0,0 +1,450 @@
+
+
+added 310 packages, and audited 311 packages in 12s
+
+44 packages are looking for funding
+  run `npm fund` for details
+
+found 0 vulnerabilities
+Finished installing dependencies
+
+Your new project is ready to go! ✨
+
+To perform an initial deployment, run `pulumi up`
+```
+
+Head to Pulumi Cloud Console and create a new Pulumi ESC project with the name `eso-esc-demo:
+
+![Pulumi ESC Environment](env1.png)
+
+And the environment `eso-esc-dev`:
+
+![Pulumi ESC Environment](env2.png)
+
+In the editor add the following yaml into the `Environment definition`:
+
+```yaml
+values:
+  pulumiConfig:
+    pulumi-pat:
+      fn::secret: <your-pulumi-pat>
+```
+
+{{% notes type="info" %}}
+
+You don't know how to create a PAT? Check out the official
+Pulumi [documentation](https://www.pulumi.com/docs/pulumi-cloud/access-management/access-tokens/).
+
+{{% /notes %}}
+
+Now, we need to link the Pulumi ESC project to the Pulumi IaC project. To do this, we need to add the following to your
+`Pulumi.dev.yaml`:
+
+```yaml
+environment:
+- eso-esc-demo/eso-esc-dev
+```
+
+After all wiring is done, we can deploy the ESO into the Kubernetes cluster. Open the `index.ts` file and add the
+following code:
+
+```typescript
+import * as k8s from "@pulumi/kubernetes";
+import * as pulumi from "@pulumi/pulumi";
+
+const config = new pulumi.Config();
+
+// Create a Kubernetes namespace
+const ns = new k8s.core.v1.Namespace("external-secrets", {
+    metadata: {
+        name: "external-secrets",
+    },
+});
+
+// Deploy a Helm release into the namespace
+const externalSecrets = new k8s.helm.v3.Release("external-secrets", {
+    chart: "external-secrets",
+    version: "0.10.4", // Specify the version of the chart
+    namespace: ns.metadata.name,
+    repositoryOpts: {
+        repo: "https://charts.external-secrets.io",
+    },
+});
+
+// Deploy a secret into the namespace
+const mySecret = new k8s.core.v1.Secret("my-secret", {
+    metadata: {
+        namespace: ns.metadata.name,
+        name: "pulumi-access-token",
+    },
+    stringData: {
+        "PULUMI_ACCESS_TOKEN": config.require("pulumi-pat"),
+    },
+    type: "Opaque",
+});
+
+// Export the namespace name
+export const namespaceName = ns.metadata.name;
+
+// Export the Helm release name
+export const helmReleaseName = externalSecrets.name;
+```
+
+Deploy the stack by running:
+
+```bash
+pulumi up
+```
+
+And you should see that the secret was created in the Kubernetes cluster and the ESO was deployed successfully.
+
+```bash
+kubectl get secret -n external-secrets pulumi-access-token -o jsonpath='{.data.PULUMI_ACCESS_TOKEN}' | base64 -d
+```
+
+### Step 2: Create an External Secret
+
+Now, we will create an external secret in the Pulumi ESC project and synchronize it into the Kubernetes cluster. To do
+this, we need to create a new secret in the Pulumi ESC project and reference it in the ESO.
+
+Create a new environment called `esp-esc-app` in the Pulumi ESC project with the name `eso-esc-demo`:
```

Comment:
...->"Create a new ESC environment called `esp-esc-app` in the `eso-esc-demo` ESC project."